qiskit/circuit/library/__init__.py

```diff
@@ -208,42 +318,79 @@
    UCRYGate
    UCRZGate
 
+The above objects derive :class:`.Gate` or :class:`~.circuit.Instruction`, which allows the
+compiler to reason about them on an abstract level. We therefore suggest using these instead
+of the following which derive :class:`.QuantumCircuit` and are eagerly constructed.
+
+.. autosummary::
+   :toctree: ../stubs/
+   :template: autosummary/class_no_inherited_members.rst
+
+   Diagonal
+   MCMT
+   MCMTVChain
+   Permutation
+   GMS
+   GR
+   GRX
+   GRY
+   GRZ
+
+.. _boolean-logic:
+
 Boolean Logic Circuits
 ======================
 
-These are :class:`~qiskit.circuit.QuantumCircuit` subclasses
-that implement boolean logic operations, such as the logical
+These :class:`.Gate`\ s implement boolean logic operations, such as the logical
 or of a set of qubit states.
 
+.. autosummary::
+   :toctree: ../stubs/
+
+   AndGate
+   OrGate
+   BitwiseXorGate
+   random_bitwise_xor
+   InnerProductGate
+
+The above objects derive :class:`.Gate` (or return this type), which allows the
+compiler to reason about them on an abstract level. We therefore suggest using these instead
+of the following which derive :class:`.QuantumCircuit` and are eagerly constructed.
 
 .. autosummary::
    :toctree: ../stubs/
    :template: autosummary/class_no_inherited_members.rst
 
    AND
-   AndGate
    OR
-   OrGate
    XOR
-   BitwiseXorGate
-   random_bitwise_xor
    InnerProduct
-   InnerProductGate
 
+.. _basis-change:
 
 Basis Change Circuits
 =====================
 
-These circuits allow basis transformations of the qubit states. For example,
+These gates allow basis transformations of the qubit states. For example,
```

Comment:
```suggestion
These gates enable basis transformations of the qubit states. For example,
```

Reply:
maybe perform? The original `allow` was not quite accurate haha 
```suggestion
These gates perform basis transformations of the qubit states. For example,
```